src/compile/mark/mixins.ts

```diff
@@ -121,7 +123,7 @@ export function bandPosition(fieldDef: FieldDef<string>, channel: 'x'|'y', model
   const scaleName = model.scaleName(channel);
   const sizeChannel = channel === 'x' ? 'width' : 'height';
 
-  if (model.encoding.size) {
+  if (model.encoding.size || model.markDef.size !== undefined) {
```

Comment:
Why are you just checking for `model.encoding.size` and not `model.encoding.size !== undefined`?

Reply:
`markDef.size` is number so `0` means something

`encoding.size` can only be an object.

Reply:
I see. I'd be okay with checking against undefined in both cases.